### PR TITLE
fix: text kwargs no longer fail with empty string

### PR DESCRIPTION
**Pull request by MthwRobinson**
### Summary

Adds a fix to partition functions that accepts a `text` kwarg. They no longer fail if an empty string is passed, but rather return an empty list of elements. Additional, `partition_json` was updated so that it does not fail if the input is an empty list. Instead, an empty list of elements is returned.

### Testing

```python
from unstructured.partition.text import partition_text

partition_text(text="")
```

```python
from unstructured.partition.json import partition_json

partition_json(text="[]")
```

### CODE REVIEW

**Review by qued — APPROVED** [review]
LGTM!

File: unstructured/partition/json.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from unstructured.partition.common import exactly_one
 from unstructured.staging.base import dict_to_elements
 
-LIST_OF_DICTS_PATTERN = r"\A\s*\[\s*{"
+LIST_OF_DICTS_PATTERN = r"\A\s*\[\s*{?"
```
Comment (cragwolfe) [inline unstructured/partition/json.py:9]: why this change?

Comment (MthwRobinson) [reply unstructured/partition/json.py:9]: It raised an error on empty lists before
